src/Xamarin.Android.Build.Tasks/Utilities/XAJavaTypeScanner.cs

```diff
@@ -0,0 +1,167 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.IO;
+//using System.IO.MemoryMappedFiles;
+
+using Java.Interop.Tools.Cecil;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Mono.Cecil;
+using Xamarin.Android.Tools;
+
+namespace Xamarin.Android.Tasks;
+
+class JavaType
+{
+	public readonly TypeDefinition Type;
```

Comment:
`Type` and `PerAbiTypes` confuses me (likely because I haven't fully read this PR yet); what assembly does `Type` come from?  Why does `Type` exist at all?

It feels like `JavaType` should "really" be an "enum class" (something C# should borrow from Swift): it's _either_ `Type`, or it's `PerAbiTypes`.  Both at the same time?  How's that work?

Reply:
`Type` is just a convenient way to access a type for code that doesn't care about per-rid differences. The assumption is that **all** the per-RID types are identical as far as their "identity" is concerned (i.e. that `Some.Namespace.Klass` is the same in any of the per-RID assemblies, **conceptually** and **API-wise**) and some code (like JCWs) needs only that "identity" information, the type name, methods etc. Other code, like typemaps or marshal methods, need the metadata info which can differ between RIDs (MVIDs, type and method tokens for instance).  So `Type` is **any** (usually the first added) `TypeDefinition` to the `PerAbiTypes` dictionary, making it easier for JCW to just use what it needs without messing with dictionaries.